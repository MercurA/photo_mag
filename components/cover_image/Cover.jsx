import Image from 'next/image';
import styles from './cover.module.css'
import SideMenu from '../menu/SideMenu';
import Comingsoon from '../misc/Comingsoon';

const Cover = ({image}) => {
    return (
        <div className={styles.container}>
            <div className={styles.effect}>
                {image ? 
                    <Image 
                    src={image?.path} 
                    fill 
                    sizes="(max-width: 1024px) 70vw , 100vw" 
                    alt={''}
                    style={{
                        objectFit: 'contain',
                        padding: '40px',
                        
                    }}
                    id='cover_image'
                /> : <Comingsoon />}
            </div>
            <SideMenu/>
        </div>
    )
}

export default Cover;