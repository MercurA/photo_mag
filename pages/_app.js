import '../styles/globals.css'
import { createContext, useReducer } from "react"
import reducer, { initialState } from "../utils/state/reducer"

export const AppContext = createContext(null)

const MyApp = ({ Component, pageProps }) => {
  const [state, dispatch] = useReducer(reducer, initialState)

  return (
    <AppContext.Provider value={{ ...state, dispatch }}>
      <Component {...pageProps} />
    </AppContext.Provider>
  )
}

export default MyApp