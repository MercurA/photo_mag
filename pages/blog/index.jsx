import DesktopBlogContainer from '../../containers/DesktopBlogContainer'

const Blog = () => {
    return (
        <DesktopBlogContainer />
    )
}

export default Blog;