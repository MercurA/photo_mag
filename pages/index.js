import { createContext, useReducer } from "react";
import CardContainer from "../components/card_image/CardContainer";
import Menu from "../components/menu/Menu";
import MusicContainer from "../components/music/MusicContainer";
import reducer, { initialState } from "../utils/state/reducer";

export const AppContext = createContext(null)

export default function Home() {
  const [state, dispatch] = useReducer(reducer, initialState)

  return (
    <AppContext.Provider value={state}>
      <div >
        <Menu dispatch={dispatch}/>
        <CardContainer />
        <MusicContainer dispatch={dispatch}/>
      </div>
    </AppContext.Provider>
  )
}
