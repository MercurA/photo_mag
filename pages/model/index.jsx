import { useRef, useState } from 'react'
import styles from './model.module.css'
import { Canvas, useFrame } from '@react-three/fiber'

const Sphere = ({position, size, color}) => {
    const [hovered, setHovered] = useState(false)
    const sphereRef = useRef()
    useFrame((state, delta) => {
        sphereRef.current.rotation.y += delta/6
    })
    return (
        <mesh ref={sphereRef} onPointerOver={(e) => {e.stopPropagation(); setHovered(!hovered)}}>
            <sphereGeometry />
            <meshStandardMaterial color={hovered ? color : 'lightblue'} wireframe/>
        </mesh>
    )
}

const Model = () => {
    return (
        <div className={styles.container}>
            <Canvas >
                <directionalLight position={[0, 0, 7]} />
                <Sphere position={[0,0,0]} size={[1,1,1]} color={'orange'}/>
            </Canvas>
        </div>
    )
}

export default Model