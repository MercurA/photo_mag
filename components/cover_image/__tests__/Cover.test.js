import {render} from '@testing-library/react'
import Cover from '../Cover'
import { AppContext } from '../../../pages/_app'

describe("Cover component", () => {
    let state = {
        currentImage: {
            path: '/images/2.png',
            alt: 'mountain view'
        }
    }
    it('should render correctly', () => {
        const comp = render(
            <AppContext.Provider value={state}>
                <Cover image={state.currentImage}/>
            </AppContext.Provider>
        )

        expect(comp).toMatchSnapshot()
    })
})