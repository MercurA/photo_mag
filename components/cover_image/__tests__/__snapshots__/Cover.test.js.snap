// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cover component should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="effect"
        >
          <img
            alt=""
            data-nimg="fill"
            decoding="async"
            id="cover_image"
            src=""
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: contain; color: transparent; padding: 40px;"
          />
        </div>
        <div
          class="container"
        >
          <div
            class="brand"
          >
            HC
          </div>
          <div
            class="categoryGroup"
          >
            <div
              class="text textUnblur"
              data-index="0"
              data-tag=""
            >
              Black & White
            </div>
            <div
              class="text "
              data-index="1"
              data-tag=""
            >
              Landscape
            </div>
            <div
              class="text "
              data-index="2"
              data-tag=""
            >
              Journalism
            </div>
          </div>
          <div
            class="contact"
          >
            <p>
              constandacheh@gmail.com
            </p>
            <div>
              <a
                class="link"
                href="https://www.instagram.com/cypher_mercurius/"
                target="_blank"
              >
                <img
                  alt="instagram logo"
                  data-nimg="1"
                  decoding="async"
                  height="20"
                  loading="lazy"
                  src="/icons/insta_white.svg"
                  style="color: transparent;"
                  width="20"
                />
              </a>
              <a
                class="link"
                href="https://github.com/MercurA"
                target="_blank"
              >
                <img
                  alt="github logo"
                  data-nimg="1"
                  decoding="async"
                  height="20"
                  loading="lazy"
                  src="/icons/github_white.svg"
                  style="color: transparent;"
                  width="20"
                />
              </a>
              <a
                class="link"
                href="https://soundcloud.com/dionis5"
                target="_blank"
              >
                <img
                  alt="github logo"
                  data-nimg="1"
                  decoding="async"
                  height="20"
                  loading="lazy"
                  src="/_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=48&q=75"
                  srcset="/_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=32&q=75 1x, /_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=48&q=75 2x"
                  style="color: transparent;"
                  width="20"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="effect"
      >
        <img
          alt=""
          data-nimg="fill"
          decoding="async"
          id="cover_image"
          src=""
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: contain; color: transparent; padding: 40px;"
        />
      </div>
      <div
        class="container"
      >
        <div
          class="brand"
        >
          HC
        </div>
        <div
          class="categoryGroup"
        >
          <div
            class="text textUnblur"
            data-index="0"
            data-tag=""
          >
            Black & White
          </div>
          <div
            class="text "
            data-index="1"
            data-tag=""
          >
            Landscape
          </div>
          <div
            class="text "
            data-index="2"
            data-tag=""
          >
            Journalism
          </div>
        </div>
        <div
          class="contact"
        >
          <p>
            constandacheh@gmail.com
          </p>
          <div>
            <a
              class="link"
              href="https://www.instagram.com/cypher_mercurius/"
              target="_blank"
            >
              <img
                alt="instagram logo"
                data-nimg="1"
                decoding="async"
                height="20"
                loading="lazy"
                src="/icons/insta_white.svg"
                style="color: transparent;"
                width="20"
              />
            </a>
            <a
              class="link"
              href="https://github.com/MercurA"
              target="_blank"
            >
              <img
                alt="github logo"
                data-nimg="1"
                decoding="async"
                height="20"
                loading="lazy"
                src="/icons/github_white.svg"
                style="color: transparent;"
                width="20"
              />
            </a>
            <a
              class="link"
              href="https://soundcloud.com/dionis5"
              target="_blank"
            >
              <img
                alt="github logo"
                data-nimg="1"
                decoding="async"
                height="20"
                loading="lazy"
                src="/_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=48&q=75"
                srcset="/_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=32&q=75 1x, /_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=48&q=75 2x"
                style="color: transparent;"
                width="20"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
